reputation/microeconomic_reputation.tex

```diff
@@ -0,0 +1,367 @@
+present the topic and conclusions while maintaining resonable level of brievity.
+
+\subsection{Abstract}
+This document provides a simple microeconomic model of the Golem marketplace. 
+Following questions are answered within the model:
+
+\begin{itemize}
+\item What is the exact problem we hope to solve with the reputation system?
+\item What do we exactly mean by the "reputation"?
+\item What are the other, non-reputation approaches to the main problem?
+\item How to measure the quality of our solutions? I.e. how to know if we succeeded?
+\item How to split the "reputation" into separate subproblems?
+\end{itemize}
+
+The topic is approached from the highest point of view - no specific solutions are proposed, only general classes of solutions are discussed.
+With an exception of a few details/examples this document describes just a "general" marketplace (replace "provider/requestor" with "seller/buyer").
+
+\subsection{Definitions}
+
+\textbf{Agent} - A decision-making entity (person, company, etc.).
+\newline
+\textbf{Utility} - The total happiness of an agent.
+\newline
+\textbf{Utility function $U$} - A function $U: StateOfTheWorld -> \mathbb{R}$, defined for a particular agent, that represents their utility in a given situation.
+\newline
+\textbf{Expected utility} - For every possible decision an agent can make ($D$) and every possible state of the world $w$, there is some probability $P: (D, w) \rightarrow [0, 1]$  that given decision will lead to the given state of the world.
+Agents don't have the full knowledge about $P$, they know only some information $I$ and an estimation of $P$ based on this information: $P_I: (D, w) \rightarrow [0, 1]$. The expected utility $E: (I, U, D) -> \mathbb{R}$ is defined as follows:
+
+\begin{equation}
+E(I, U, D) = \sum_{w \in all\_possible\_world\_states}U(w) * P_I(D, w)
+\end{equation}
+
+We'll be also using a shorter notation, $E(I, U, D) = E_A(D)$ to describe "expected utility of the decision $D$, for agent A, who has information $I$ and utility function $U$".
+
+In other words, for agent $A$ and decision $D$, $E_A(D)$ is "how happy agent $A$ expects to be if they do $D$".
+\newline
+
+\textbf{Golem} - The observable result of all the work done by the Golem Factory - the Golem protocol, available software, state of the market etc.
+\newline
+
+\subsection{Additional assumptions}
+
+\begin{enumerate}
+\item The Expected Utility Hypothesis\footnote{\href{https://en.wikipedia.org/wiki/Expected\_utility\_hypothesis}{https://en.wikipedia.org/wiki/Expected\_utility\_hypothesis}}: every agent tries to maximize their expected utility.
+
+\item Every agent can costlessy access and analyze all of the publicly available information about Golem. Thanks to this assumption we can remove $I$ from the equations - we no longer
+    care about "what agent knows", but only about "what really is". This is a major simplification that might not be a good approximation of the "real world" Golem market, but we
```

Comment:
I am not convinced by this argument. The mere fact that there is no cost of accessing the information does not mean the information itself has no value. In fact asserting that would undermine the whole point of a reputation system.

Reply:
Good point. What I meant was:
* The general information about Golem is free, e.g. things like "how efficient is `yagna`", "what are the bugs/vulnerabilities in the provided software", "how many providers they are and how much work they do", "what is the average price variability"
* The information regarding **particular agents** is not free (and not really available, and increasing the avaiability is a part of the reputation system)

But that's what I meant, not what I wrote :)
I will fix this. Thx.

Reply:
Fixed in ac044d4 (a footnote)

Reply:
I still think this assumption is suspicious on fundamental grounds as well as counterproductive from the pragmatic POV. For example it seems to exclude powerful arguments along the lines of "Golem value with information is bigger than without it, thus the value of the information is such and such."

Reply:
I don't understand. 

Let's say the information is "what is the chance the given runtime will not start because of the `yagna` bug". What do we gain by analyzing the value of this information, or by considering an agent who has an incorrect estimation of it?